README.md

```diff
@@ -1,68 +1,118 @@
     2. `cd poa-bridge-contracts`
-    3. Checkout branch `v2` : `git checkout v2`
+    3. Checkout branch `refactor_v1` : `git checkout refactor_v1`
     4. Compile contracts: `truffle compile`
     5. Go to deploy folder: `cd deploy`
     6. create a `.env` file: `cp .env.example .env` (look at `.env.example` to see the variables that need to be present)
     7. Execute `node deploy.js`
 
-2. Install [RabbitMQ](https://www.rabbitmq.com/) and [Redis](https://redis.io/)
-  - RabbitMQ version: `3.7`
-  - Redis version: `4.0`
-  
-3. Create a `.env` file: `cp .env.example .env` and fill in the information using the output data from previous deploy step. Check the `.env.example` file to see the required variables.
+2. Start Redis and RabbitMQ in your local environment.
+    1. `redis-server` starts Redis. `redis-cli ping` will return a `pong` if Redis is running.
+    2. `rabbitmq-server` starts RabbitMQ. Use `rabbitmqctl status` to check if RabbitMQ is running. 
+
+3. Create a `.env` file: `cp .env.example .env`
+   1. fill in the required information using the output data from the bridge contract deployment. Check the `.env.example` file to see the required variables.
 
-## Run the processes
+## Run the Processes
 
-### Native to Erc mode
+### Native to ERC mode
   - `npm run watcher:signature-request`
   - `npm run watcher:collected-signatures`
   - `npm run watcher:affirmation-request`
   - `npm run sender:home`
   - `npm run sender:foreign`
 
-To send deposits to home contract run `node tests/sendUserTxToHome.js`
+To send deposits to a home contract run `node scripts/sendUserTxToHome.js`
 
-To send withdrawals to foreign contract run `node tests/sendUserTxToForeign.js`
+To send withdrawals to a foreign contract run `node scripts/sendUserTxToForeign.js`
 
-Make sure your `HOME_MIN_AMOUNT_PER_TX` and `FOREIGN_MIN_AMOUNT_PER_TX` is same as in your .env deployment contract
+Make sure your `HOME_MIN_AMOUNT_PER_TX` and `FOREIGN_MIN_AMOUNT_PER_TX` is the same as in your .env deployment contract.
 
 
-### Erc to Erc mode
+### ERC to ERC Mode
 
-On `.env` file set `BRIDGE_MODE=ERC_TO_ERC`
+In `.env` file set `BRIDGE_MODE=ERC_TO_ERC`
 
   - `npm run watcher:signature-request`
   - `npm run watcher:collected-signatures`
   - `npm run watcher:affirmation-request`
   - `npm run sender:home`
   - `npm run sender:foreign`
 
-To deposit from Foreign to Home contract run `node scripts/sendUserTxToErcForeign.js 10` where `10` is how many tx you would like to send out
+To deposit from Foreign to Home contract run `node scripts/sendUserTxToErcForeign.js 10` where `10` is how many tx you would like to send out.
 
-To withdrawal to Home to Foreign contract run `node scripts/sendUserTxToErcHome.js 10` where `10` is how many tx you would like to send out
+To withdrawal to Home to Foreign contract run `node scripts/sendUserTxToErcHome.js 10` where `10` is how many tx you would like to send out.
 
```

Comment:
Consider to have a separate section describing how to rollback "last processed block" in Redis DB.

It is needed in cases if it was found that by some reason the bridge does handle some events properly. E.g. a transaction was hanged due to low gas price. If so, it is necessary to identify which watcher needs to re-run (e.g. if the validator signatures were collected but the transaction with signatures was not sent to the Foreign network, the `collected-signatures` watcher must look at the block where the corresponding `CollectedSignatures` event was raised) and the following command should be executed in the bridge root directory:

```shell
bash ./reset-lastBlock.sh <watcher> <block num>
```
where _watcher_ could be one of:
  * `signature-request`
  * `collected-signatures`
  * `affirmation-request`